lib/AST/ASTDumper.cpp

```diff
@@ -656,9 +656,14 @@ namespace {
 
     void visitAssociatedTypeDecl(AssociatedTypeDecl *decl) {
       printCommon(decl, "associated_type_decl");
-      if (auto defaultDef = decl->getDefaultDefinitionType()) {
-        OS << " default=";
-        defaultDef.print(OS);
+
+      OS << " default=";
+      if (auto type = decl->getCachedDefaultDefinitionType()) {
+        type->print(OS);
+      } else if (auto repr = decl->getDefaultDefinitionTypeRepr()) {
+        repr->print(OS);
+      } else {
+        OS << "<not computed>";
```

Comment:
"Not computed" and "nonexistent" are different things here. Can we differentiate in the AST dumper output?